ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -814,6 +814,11 @@ object Trees {
     val tpe = NoType
   }
 
+  case class TopLevelModuleExportDef(fullName: String)(
+      implicit val pos: Position) extends Tree {
+    val tpe = NoType
+  }
+
```

Comment:
Have you considered giving the `ModuleExportDef` a `lazy` flag instead? IIUC, we eventually want to get rid of lazy module exports (i.e. module exports being called by function) and only allow those (or export an accessor function). We might make this more explicit by giving `ModuleExportDef` a `lazy` flag and hopefully making the parallel to constructor export def more clear.

In any case, please document the difference between these, since as it stands, it is impossible to understand the difference between them when only looking at `Trees.scala`.